### PR TITLE
kubeadm: try to reuse `RunData` if possible

### PR DESCRIPTION
**Pull request by chendave**
The generation of `RunData` for init and join is relative time consuming, the data should be shared / reused whenever possible.

/sig cluster-lifecycle   

Signed-off-by: Dave Chen <dave.chen@arm.com>

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by chendave** [comment]
/kind feature ? (For it intends to improve the performance)

**Review by neolit123 — COMMENTED** [review]
looks like an ok change to me.
would like another pair of eyes to double check.

/approve

File: cmd/kubeadm/app/cmd/phases/workflow/runner.go [hunk 1]
```diff
@@ -222,9 +222,10 @@ func (e *Runner) Run(args []string) error {
 	}
 
 	// builds the runner data
-	var data RunData
-	if data, err = e.InitData(args); err != nil {
-		return err
+	if data == nil {
```
Comment (pacoxu) [inline cmd/kubeadm/app/cmd/phases/workflow/runner.go:225]: Why not use `e.runData` here?

Comment (chendave) [reply cmd/kubeadm/app/cmd/phases/workflow/runner.go:225]: yes, that would be simpler, thanks!

**Pull request merged**